Enable mosaic creation

vgeorge opened the pull request:
Contributes to #22. 

Current status:

- Added a new section to create mosaics
- Added formatted UTC dates to the existing mosaics
- Enabled button for creating mosaics in the API but due to https://github.com/developmentseed/pearl-backend/issues/74 it is not possible to run predictions yet

To do:

- [x] Add tabs 'Preset Mosaics' and 'Create Mosaics'
- [x] Add mosaic preview before creation
- [ ] Add cloud cover slider

@LanesGood would you be able to look into adding the tabs? I tried to use `TabbedBlock`, but I couldn't figure out how to set the container height using this component.

vgeorge commented:
I forgot to mention this is based on #67 and I set its branch as the target for this PR.

LanesGood commented:
@vgeorge I set a fixed height and fixed the overflow text for the mosaic modal. It seems the tabs are not activated on click; I do think we're running into some issues due to the `TabbedBlock` component being designed for use within the `PrimePanel`. I think we can fix these issues without introducing another library, but up to you:
<img width="1440" alt="image" src="https://github.com/developmentseed/pearl-frontend/assets/12634024/847360e6-0219-4a8c-9a39-ff1169491457">

vgeorge commented:
@LanesGood I fixed the tab switch and also enabled creating mosaics in the API. The next step for me is to add a map preview.

vgeorge commented:
I did the following changes:

- Added a map preview
- Updated the form to be more similar to the Figma design

Known issues:

- Left alignment inside the modal, styles of mosaic creation form (@LanesGood can you please try fixing those?)
- No filtering: in our last sync with Reforestamos it didn't became clear that they needed it, we can add those in separate once we have more clarity

@LanesGood @geohacker you should be able to create and use custom mosaics from PC, this is ready for a first functionality review.

LanesGood commented:
Styled modal & form:
<img width="1248" alt="image" src="https://github.com/developmentseed/pearl-frontend/assets/12634024/9d52b26c-b99f-4207-834a-fb55839499cd">

@vgeorge I do see that selecting NAIP is allowing mosaics. I didn't see this in previous branches. Is this behavior intended?
<img width="1248" alt="image" src="https://github.com/developmentseed/pearl-frontend/assets/12634024/31fb7c14-07cc-4408-ae43-1c362e73cc5d">

vgeorge commented:
@LanesGood the explorer on Planetary Computer uses 10% of maximum cloud cover, which I applied now. I also reduced the width of the custom mosaic form, which seems too wide, but feel free to adjust it differently. About NAIP, we need to disable mosaic creation for it, but I would prefer to work on the issue on a different PR as it might require changes in the API.

geohacker commented:
This looks really good @vgeorge @LanesGood 👏  Only one thing I notice is that when I scroll the date picker, it seems to send multiple register requests to Microsoft. Could we bind that to the date select mouse click event or so to avoid creating many mosaic requests?

vgeorge commented:
@geohacker thanks for the review! The date input behavior looks okay to me. It sends a new request only when the date is changed, which can happen if you click on a date or change months/years by using the buttons. If you just scroll with the mouse without effectively changing the date, no requests are triggered. Please refer to this capture I did on Chrome; the scrolling happens at the beginning with no requests:

![requests](https://github.com/developmentseed/pearl-frontend/assets/329694/77122707-907c-47f7-a611-749950a259f8)

I'll go ahead and merge this, let's open in a separate ticket if you were referring to a different behavior.

geohacker commented:
@vgeorge Ah I see, yes that makes sense. Thank you!
